tests/snippets/builtin_all.py

```diff
@@ -6,5 +6,5 @@
 assert all([])
 assert not all([False, TestFailingBool()])
 
-assert_raises(RuntimeError, lambda: all(TestFailingIter()))
```

Comment:
We also could refactor this into:

```python
with assert_raises(RuntimeError):
    all(TestFailingIter())
```

What is the benefit of the lambda function type of assert_raises? In my opinion, the form with the `with` statement is clearer than the lambda-style.

What is your opinion on this? Should we go for one of the two styles?

Reply:
I agree that using lambda doesn't have much advantages for single case. I prefer to have single line test code to copy and edit many similar cases - which is common in test codes.

I don't have a nice answer about this, but maybe CPython code can give you some hints. I often can find this patterns in CPython test codes too. In this case, usually the lambda body is very short.